include/Lv2Basics.h

```diff
@@ -37,6 +37,12 @@
 namespace lmms
 {
 
+template<class T>
+struct LilvPtrDeleter
+{
+	void operator()(T* n) { lilv_free(static_cast<void*>(n)); }
+};
+
```

Comment:
I don't think this needs to be a class template. If the operator takes `void*`, it should still work with `std::unique_ptr`.

Reply:
A `void*` would suffice, but then, in statements like
```c++
auto bundlePath = AutoLilvPtr<char>(lilv_file_uri_parse(bundleUri, NULL));
```
we would lose the information that this is a `char` pointer. So, to improve type safety, I think a template makes sense.

Reply:
No, you can have a `std::unique_ptr<T>` that stores `T*` but uses a deleter which takes `void*`. This works because `T*` implicitly converts to `void*`.

Here's a working demo: https://godbolt.org/z/Msfrc5jfa

Reply:
True, but if we get an explicit cast for free, I prefer that.